mypy/semanal.py

```diff
@@ -4170,6 +4171,12 @@ def lookup_current_scope(self, name: str) -> Optional[SymbolTableNode]:
         else:
             return self.globals.get(name)
 
+    def current_scope_identifier(self) -> Optional[str]:
```

Comment:
This is a little inconsistent with how we track scopes elsewhere. It would be better to mirror the logic used for the `locals` attribute (also `block_depth` a few others). We'd maintain a stack and append a fresh set on each `enter()` or `enter_class()` call, and pop an item on each `leave()` or `leave_class()` call. We can then use `missing_names[-1]` to access the innermost scope.

Reply:
That makes a lot more sense, cheers for the hint!